lib/commands/skill/add-locales/helper.js

```diff
@@ -0,0 +1,148 @@
+    _getNewLocaleModelUri(selectedLocales, profile, doDebug, (resolveErr, iModelSourceByLocales) => {
+        if (resolveErr) {
+            return callback(resolveErr);
+        }
+        const uniqueTemplateSet = new Set(iModelSourceByLocales.values());
+        const iModelFolderPath = path.join(process.cwd(), ResourcesConfig.getInstance().getSkillMetaSrc(profile),
+            CONSTANTS.FILE_PATH.SKILL_PACKAGE.INTERACTION_MODEL, 'custom');
+        // copy iModel JSONs that contains same language
+        iModelSourceByLocales.forEach((v, k) => {
+            if (fs.existsSync(v)) {
+                const sourceLocale = path.basename(v, path.extname(v));
+                const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+                fs.copySync(v, targetFilePath); // do not fail if the target locale exists already
+                uniqueTemplateSet.delete(v);
+                const sourceManifestLocale = Manifest.getInstance().getPublishingLocale(sourceLocale);
+                Manifest.getInstance().setPublishingLocale(k, sourceManifestLocale);
+            }
+        });
+        // retrieve remote templates when no existing JSON shares the same language
+        _retrieveTemplatesByLanguage(uniqueTemplateSet, doDebug, (templateErr, templateByLanguage) => {
+            if (templateErr) {
+                return callback(templateErr);
+            }
+            iModelSourceByLocales.forEach((v, k) => {
+                if (!fs.existsSync(v)) {
+                    const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+                    fs.writeFileSync(targetFilePath, templateByLanguage.get(v), { encoding: 'utf-8' });
+                    Manifest.getInstance().setPublishingLocale(k, { name: 'please change' });
+                }
+            });
+            Manifest.getInstance().write();
+            callback(undefined, iModelSourceByLocales);
+        });
+    });
+}
+
+/**
+ * Based on user's selected locales, and the local project's iModel files, decide where to get the
+ * iModel file to add in user's local project. Exclude reusableLocales themselves from the result.
+ *
+ * @param {Array} selectedLocales
+ * @param {String} profile
+ * @param {Boolean} doDebug
+ * @returns {Function} callback Map { locale: filePath/remoteURI }
+ */
+function _getNewLocaleModelUri(selectedLocales, profile, doDebug, callback) {
+    const skillMetaController = new SkillMetadataController({ profile, doDebug });
+    const localIModelByLocale = skillMetaController.getInteractionModelLocales();
+    const result = new Map();
+    // get from map of templates
+    httpClient.request({
+        url: CONSTANTS.TEMPLATES.INTERACTION_MODEL_MAP,
+        method: CONSTANTS.HTTP_REQUEST.VERB.GET
+    }, 'GET_INTERACTION_MODEL_MAP', doDebug, (error, templateMapResponse) => {
+        if (error || !templateMapResponse.statusCode || templateMapResponse.statusCode !== 200) {
```

Comment:
not having status code and being error probably should be handled in the http client logic

Reply:
i will add the case when statusCode not exist into httpClient. but I feel we should not have logic to check statusCode as it's up to the upstream to decide - there might be the case when certain special statusCode results in different handling logic.